corehq/mobile_ucr_v2_update_script.py

```diff
@@ -0,0 +1,110 @@
+from corehq.util.log import with_progress_bar
+
+
+PROCESSED_DOMAINS_PATH = '/home/zandre/cchq/updated_domains.ndjson'
+LOG_FILE = '/home/zandre/cchq/update_to_v2_ucr_script.log'
+
+V1_FIXTURE_IDENTIFIER = 'src="jr://fixture/commcare:reports'
+V1_FIXTURE_PATTERN = r'<.*src="jr://fixture/commcare:reports.*>'
+V1_REFERENCES_PATTERN = r"<.*instance\('reports'\)/reports/.*>"
+RE_V1_ALL_REFERENCES = re.compile(f"{V1_FIXTURE_PATTERN}|{V1_REFERENCES_PATTERN}")
+
+skip_domains = set()
+
+
+def process(dry_run=False):
+    try:
+        processed_domains = read_ndjson_file(PROCESSED_DOMAINS_PATH)
+    except FileNotFoundError:
+        processed_domains = set()
+
+    mobile_ucr_domains = find_domains_with_toggle_enabled(MOBILE_UCR)
+
+    save_in_log(f"Number of domains with mobile ucr flag enabled: {len(mobile_ucr_domains)} ")
+
+    for domain in with_progress_bar(mobile_ucr_domains):
+        if domain in processed_domains:
+            save_in_log(f"Already processed domain: {domain}")
+            continue
+        if domain in skip_domains:
+            save_in_log(f"Skipped domain: {domain}")
+            continue
+
+        save_in_log(f"Processing domain: {domain} ...")
+        app_ids = list(get_latest_app_ids_and_versions(domain))
+        apps = get_apps_by_id(domain, app_ids)
+        for app in apps:
+            try:
+                # Don't look at app.is_released since the latest version might not be released yet
+                if app.mobile_ucr_restore_version != '2.0':
+                    save_in_log(f"Processing App: {domain}: {app.name}: {app.id}")
+                    if not has_non_v2_form(domain, app):
+                        update_app(domain, app, dry_run)
+                    else:
+                        save_in_log(
+                            f"App contains V1 references and couldn't updated: {domain}: {app.name}: {app.id}",
+                        )
+            except Exception as e:
+                save_in_log(f"Error occurred for {domain}: {str(e)}")
+                save_in_log(traceback.format_exc())
+                continue
+        save_as_ndjson(PROCESSED_DOMAINS_PATH, domain)
+
+
+def save_in_log(data):
+    print(data)
+    with open(LOG_FILE, 'a') as file:
+        file.write(data + '\n')
+
+
+def save_as_ndjson(path, data):
+    with open(path, 'a') as file:
+        print(json.dumps(data, separators=(',', ':')), file=file)
+
+
+def read_ndjson_file(path):
+    with open(path, 'r') as file:
+        return set(json.loads(line) for line in file.readlines())
+
+
+def has_non_v2_form(domain, app):
+    for form in app.get_forms():
+        save_in_log(f"Processing Form: {domain}: {form.name}")
+        # The second condition should always be False if the first one is
+        # but just as a precaution we check for it
+        if V1_FIXTURE_IDENTIFIER in form.source or RE_V1_ALL_REFERENCES.search(form.source):
+            save_in_log(f"App Contains V1 Refs: {domain}: {app.name}")
```

Comment:
Do you think we should add the form name or form id in the log as well ? 
Based on how we plan for assisting projects to update their manual references, this might be helpful. It does have a downside of checking for all the forms thereby increasing the script execution time.